src/components/table/table.interface.ts

```diff
@@ -1,3 +1,10 @@
+export interface Item {
+  selected?: boolean;
+  // TODO: Try applying proper types to this
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
```

Comment:
This is not necessary, as you can simply inherit from `Item` and then pass this type to the component?